neuromation/cli/job.py

```diff
@@ -1127,3 +1167,47 @@ def calc_statuses(status: Sequence[str], all: bool) -> Set[JobStatus]:
                 return parse_columns(format_str)
         return COLUMNS
     return format
+
+
+async def calc_life_span(client: Client, value: Optional[str]) -> Optional[float]:
+    delta = (
+        _parse_timedelta(value)
+        if value is not None
+        else await calc_default_life_span(client)
+    )
+    seconds = delta.total_seconds()
+    if seconds == 0:
+        return None
+    assert seconds > 0, f"life-span cannot be negative, got: {seconds}"
```

Comment:
Please raise `ValueError` instead of `AssertionError` here.

Reply:
It's impossible situation, and we need this check only in case in order to protect ourselves from sending negative values to the server one day if we change something in future (for example, we are going to change toml parsers). So I cannot test this check properly. Here's what we can do:
1. leave assertion
2. change to ValueError but still keep it untested
3. remove the check completely

Reply:
I read `assert expr, msg` as something that you want to expose to a user.
If the situation is impossible I expect bare `assert expr` at least. Like for the sake of `mypy` correctness, we have `assert var is not None` very often without a custom provided error message.

Reply:
I was always thinking that `assert expr` is a bad practice since if anything ever goes wrong, the user will not get any idea on the reason, just `AssertionError`.

Reply:
changed to bare `assert expr`